src/lib/y2partitioner/dialogs/partition_size.rb

```diff
@@ -10,143 +13,335 @@ module Dialogs
     # Part of {Sequences::AddPartition}.
     # Formerly MiniWorkflowStepPartitionSize
     class PartitionSize < CWM::Dialog
-      # @param disk []
-      # @param ptemplate []
-      # @param slots []
-      def initialize(disk, ptemplate, slots)
+      # @param disk_name [String]
+      # @param ptemplate [#region]
+      # @param regions [Array<Y2Storage::Region>]
+      def initialize(disk_name, ptemplate, regions)
         textdomain "storage"
-        @disk = disk
+        @disk_name = disk_name
         @ptemplate = ptemplate
-        @slots = slots
+        @regions = regions
       end
 
+      # @macro seeDialog
       def title
         # dialog title
-        Yast::Builtins.sformat(_("Add Partition on %1"), @disk.name)
+        Yast::Builtins.sformat(_("Add Partition on %1"), @disk_name)
       end
 
+      # @macro seeDialog
       def contents
-        VBox(SizeWidget.new(@disk, @ptemplate, @slots))
+        HVSquash(SizeWidget.new(@ptemplate, @regions))
+      end
+
+      # Like CWM::RadioButtons but each RB has a subordinate indented widget.
+      # This is kind of like Pager, but all Pages being visible at once,
+      # and enabled/disabled.
+      # Besides {#items} you need to define also {#widgets}.
```

Comment:
maybe add TODO to move it to yast2 CWM? I think it can be useful in other places too.